src/io/output_mesh.hh

```diff
@@ -118,6 +137,21 @@ protected:
 		discont   //!< discontinuous mesh
 	};
 
+	/**
+	 * Construct empty output mesh.
+	 *
+	 * Use in make_serial_master_mesh method and create mesh of same type as this object (continuous / discontinuos)
+	 */
+	virtual std::shared_ptr<OutputMeshBase> construct_mesh()=0;
+
+	/// Create serial (collective) nodes cache on zero process
+	virtual std::shared_ptr<ElementDataCache<double>> make_serial_nodes_cache(std::shared_ptr<ElementDataCache<unsigned int>> global_offsets)=0;
+
+	/// Create serial (collective) connectivity cache on zero process
+	virtual std::shared_ptr<ElementDataCache<unsigned int>> make_serial_connectivity_cache(std::shared_ptr<ElementDataCache<unsigned int>> global_offsets)=0;
```

Comment:
Improve doc for these methods. Note that they only implement parts of make_serial_master_mesh that are specific for continuous and discontinuous case.